File: Examples/DotNET/CSharp/Document/MergePdfFiles.cs
using System;
using Com.Aspose.PDF.Api;
using Com.Aspose.PDF.Model;
using Com.Aspose.Storage.Api;

namespace Document
{
    class MergePdfFiles
    {
        public static void Run()
        {
            // ExStart:1
            PdfApi pdfApi = new PdfApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "sample-merged.pdf";
            String storage = "";
            String folder = "";
            MergeDocuments body = new MergeDocuments();
            body.List = new System.Collections.Generic.List<string> { "sample.pdf", "input.pdf" };

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate("sample.pdf", "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + "sample.pdf"));
                storageApi.PutCreate("input.pdf", "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + "input.pdf"));

                // Invoke Aspose.PDF Cloud SDK API to merge pdf files
                DocumentResponse apiResponse = pdfApi.PutMergeDocuments(fileName, storage, folder, body);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    Console.WriteLine("Merge Multiple PDF Files, Done!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Perl/Document/MergePdfFiles.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;

use AsposePdfCloud::PdfApi;
use AsposePdfCloud::ApiClient;
use AsposePdfCloud::Configuration;
use AsposePdfCloud::Object::MergeDocuments;

my $configFile = '../config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposePdfCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposePdfCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposePdfCloud::Configuration::debug = 1;

$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Pdf API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $pdfApi = AsposePdfCloud::PdfApi->new();

# Set input file name
my $name =  'sample-merged.pdf';
my $mergefilename1 = 'Sample.pdf';
my $mergefilename2 = 'sample-input.pdf';
my @mergeDocumentsBody = AsposePdfCloud::Object::MergeDocuments->new('List' => [$mergefilename1,$mergefilename2 ]);

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);
$response = $storageApi->PutCreate(Path => $mergefilename1, file => $data_path.$mergefilename1);
$response = $storageApi->PutCreate(Path => $mergefilename2, file => $data_path.$mergefilename2);

# Invoke Aspose.Pdf Cloud SDK API to merge multiple PDF files                           
$response = $pdfApi->PutMergeDocuments(name=>$name,body=>@mergeDocumentsBody);

if($response->{'Status'} eq 'OK'){
    # Download merged document from storage server
    my $output_file = $out_path. 'merged-' . $name;
    $response = $storageApi->GetDownload(Path => $name);
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1